src/classes/Ingredient.js

```diff
@@ -1,12 +1,22 @@
+import { ingredientsData } from '../data/ingredients'
+
 class Ingredient {
   constructor(ingredient) {
     this.id = ingredient.id;
-    this.name = ingredient.name;
-    this.estimatedCostInCents = ingredient.estimatedCostInCents;
+    this.amount = ingredient.quantity.amount;
+    this.unit = ingredient.quantity.unit;
+    this.name = this.retrieveData('name');
+    this.estimatedCostInCents = this.retrieveData('estimatedCostInCents');
+    this.totalCostInDollars = this.calculateCost();
+  }
+
+  retrieveData(property) {
+    const ingredient = ingredientsData.find(ingredient => ingredient.id === this.id);
+    return ingredient[property];
   }
 
-  calculateCost(amount) {
-    return this.estimatedCostInCents * amount / 100;
+  calculateCost() {
```

Comment:
I wouldn't think this would happen, but there's a small chance that the calculateCost func will run alongside the retrieveData('estimatedCost') func, thus making the calculateCost func return undefined * this.amount, etc. I think it will be fine, but it's something to watch for when running the tests

Reply:
Good point, Paul! I built it using the tests and all the tests passed so luckily it was not an issue, but maybe something for us to look out for.